transactions/WFLY-14869-microprofile-lra.adoc

```diff
@@ -0,0 +1,118 @@
+= Support for MicroProfile LRA
+:author:            Ondrej Chaloupka
+:email:             ochaloup@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+:keywords:          transaction, saga, lra, microprofile
+
+== Overview
+
+The goal of this RFE is providing the Galleon feature pack for WildFly
+which integrates the https://github.com/jbosstm/narayana/tree/master/rts/lra[Narayana implementation]
+of the https://github.com/eclipse/microprofile-lra[MicroProfile LRA specification].
+
+The MicroProfile LRA is a transaction processing pattern well suited for microservice development
+based on the saga pattern. The motivation of such processing is summarized in the https://github.com/eclipse/microprofile-lra/blob/1.0/README.adoc#motivation[MicroProfile Spec readme].
+
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-14869
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1496
+
+=== Dev Contacts
+
+* mailto:{mkunwar@redhat.com}[Mayank Kunwar]
+* mailto:{mstefank@redhat.com}[Martin Stefanko]
+* mailto:{ochaloup@redhat.com}[{author}]
+
+=== QE Contacts
+
+* TBD
+
+=== Testing By
+
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [ ] Engineering
+* [ ] QE
+
+TBD
+
+=== Affected Projects or Components
+
+* Transactions
+* WildFly
+
+=== Other Interested Projects
+
+* MicroProfile
+* Galleon
+
+=== Relevant Installation Types
+
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+* [ ] Managed domain
+* [ ] OpenShift s2i
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+The feature pack will integrate the LRA specification as the Galleon feature pack.
+The Galleon layer will be available for provisioning.
+The provisioned feature pack will provide all functionality defined in LRA specification for users in WildFly.
+
+The feature pack will be split to two Galleon layers (i.e., two separate subsystems).
+
+The first layer provides a capability of the LRA client - an application library that provides
+CDI annotations to manage the scope of the LRA and programmatic API to manage the LRAs.
```

Comment:
I don't think we want to support a programmatic API just yet, the MP LRA spec only provides annotation support, the programmatic API won't be available until spec version 1.x. Also the client API is for our internal use only and we must not encourage users to use it.

Reply:
Absolutely correct. I removed that statement from the proposal.